internal/node/node_test.go

```diff
@@ -220,3 +269,64 @@ var _ = Describe("FindNodeCondition", func() {
 		Expect(conditionFound).To(BeNil())
 	})
 })
+var kernelModuleReadyNodeLabel = utils.GetKernelModuleReadyNodeLabel("loaded-ns", "loaded-n")
+
+var _ = Describe("addLabels", func() {
+	var (
+		n    Node
+		node v1.Node
+	)
+
+	BeforeEach(func() {
+		n = NewNode(nil)
+		node = v1.Node{}
+	})
+
+	It("Should add labels", func() {
+		labels := []string{kernelModuleReadyNodeLabel}
+
+		Expect(node.Labels).ToNot(HaveKey(kernelModuleReadyNodeLabel))
+		n.addLabels(&node, labels)
+		Expect(node.Labels).To(HaveKey(kernelModuleReadyNodeLabel))
+	})
+	It("Should not add labels", func() {
+		var labels []string
+
+		Expect(node.Labels).ToNot(HaveKey(kernelModuleReadyNodeLabel))
+		n.addLabels(&node, labels)
+		Expect(node.Labels).ToNot(HaveKey(kernelModuleReadyNodeLabel))
+	})
+})
+
+var _ = Describe("removeLabels", func() {
+	var (
+		n    Node
+		node v1.Node
+	)
+
+	BeforeEach(func() {
+		n = NewNode(nil)
+		node = v1.Node{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: map[string]string{
+					kernelModuleReadyNodeLabel: "",
+				},
+			},
+		}
+	})
+
+	It("Should remove labels", func() {
+		labels := []string{kernelModuleReadyNodeLabel}
+
+		Expect(node.Labels).To(HaveKey(kernelModuleReadyNodeLabel))
+		n.removeLabels(&node, labels)
+		Expect(node.Labels).ToNot(HaveKey(kernelModuleReadyNodeLabel))
+	})
+	It("Should not remove labels", func() {
+		var labels []string
+
+		Expect(node.Labels).To(HaveKey(kernelModuleReadyNodeLabel))
+		n.removeLabels(&node, labels)
+		Expect(node.Labels).To(HaveKey(kernelModuleReadyNodeLabel))
+	})
```

Comment:
We can remove this test - it is not really testing anything different than the previous spec.

Reply:
Same as https://github.com/kubernetes-sigs/kernel-module-management/pull/850#discussion_r1677109887.

It is not crucial at all to have it. If you think there is benefit in it then feel free to keep it. I don't see a lot of value in it but it is your decision :)

Reply:
test is removed